tests/unit/s2n_post_handshake_test.c

```diff
@@ -153,7 +141,12 @@ int main(int argc, char **argv)
 
                 EXPECT_SUCCESS(s2n_stuffer_write_uint8(&conn->in, state_machine[i].message_type));
                 EXPECT_SUCCESS(s2n_stuffer_write_uint24(&conn->in, 0));
-                EXPECT_FAILURE_WITH_ERRNO(s2n_post_handshake_recv(conn), S2N_ERR_BAD_MESSAGE);
+
+                if (state_machine[i].message_type == TLS_HELLO_REQUEST) {
+                    EXPECT_SUCCESS(s2n_post_handshake_recv(conn));
```

Comment:
I think this branch is never reached? No element of the `state_machine` array has a `message_type` of `TLS_HELLO_REQUEST`: 

https://github.com/aws/s2n-tls/blob/main/tls/s2n_handshake_io.c#L67-L85

Reply:
Good catch! Added a more explicit test, put the old test back the way I found it.